docs/users/developing.md

```diff
@@ -388,6 +388,46 @@ $ open htmlcov/index.html
 
 to see a code coverage report. This uses [Coverage.py](https://coverage.readthedocs.io/) and also requires a pytest plugin (`pytest-cov`), both of which should be installed if you are using the provided conda environments. CodeCov provides this as an automated service with their own web frontend using a similar file generated by our CI. However, it can be useful to run this locally to check coverage changes (i.e. "is this function I added sufficiently coverged by tests?") without waiting for CI to complete.
 
+### Jupyter Notebooks in Rendered Docs
+
+We sometimes include [Jupyter Notebooks] in our book-style documentation. These are files with the `.ipynb` extension. Notebooks allow sections of the docs that include a large number of code examples to be tested easily, and support automatically rendering example output, including interactive [Jupyter widgets] such as the 3D molecule representations in the [molecule cookbook]. Note that the following information only applies to notebooks in the `docs` directory; notebooks in the `examples` directory are not included in the Toolkit docs (but may be included in [OpenFF Examples] via a different process).
+
+:::{admonition} Information
+For interactive widgets to appear in rendered documentation, they must be stored in the notebook each time it is re-executed. For information on how to do this, see [](#widgets).
+:::
+
+[Jupyter Notebooks]: https://jupyter.org/
+[Jupyter widgets]: https://ipywidgets.readthedocs.io/
+[molecule cookbook]: molecule_cookbook.ipynb#from-pdb-file
+[OpenFF Examples]: https://docs.openforcefield.org/examples
+
+#### MyST Markdown
+
+Documentation notebooks are rendered with the [MyST-NB] Sphinx extension, which provides the expanded Markdown syntax used in other Markdown files in this documentation. Admonitions, maths, and links to API references and other documentation pages can be included in the usual ways. This means that some correct Markdown will appear incorrect in Jupyter by default, as it uses a simpler Markdown dialect. To render MyST syntax, you can install the [`jupyterlab-myst`] Jupyter extension:
+
+```shell
+mamba install -c conda-forge jupyterlab-myst
+```
+
+Note that some features that will render correctly in Sphinx rely on extensions that are not available to `jupyterlab-myst`, and so may still appear incorrect even with the extension.
+
+[MyST-NB]: https://myst-nb.readthedocs.io/
+[`jupyterlab-myst`]: https://mystmd.org/guide/quickstart-jupyter-lab-myst
+
+#### Widgets
+
+Jupyter Notebooks can produce interactive [widgets] as output, and MyST-NB supports rendering them in HTML output. Any given widget includes both Python code and JavaScript code, and Jupyter defines ways for them to speak to each other which allows Python code to be run interactively via the widget in the notebook. 
```

Comment:
I don't buy the logical structure of this paragraph - the second sentence introduces Python <-> JavaScript communication, but I don't see how that enables Python <-> Jupyter communication "which allows Python code to be run interactively via the widget in the notebook" as I thought that works with vanilla Python + Jupyter. Maybe I mis-understanding the contents of a widget or where the JavaScript lies?

Reply:
I'm confused about your confusion :/ I'm not really sure what distinction you're drawing between Jupyter and JavaScript. I think by Jupyter you mean, like, how its rendered in the notebook - but as far as I know Jupyter just says "OK put some JS and HTML here" and all the rendering happens in JS. The JS just sorta renders directly to the browser, it doesn't really communicate with Jupyter at all beyond being told "here's where you should put stuff". Python can do something similar by writing HTML output I think, which is rendered richly by Jupyter, but Widgets specifically go further and incorporate JS to provide interactivity, and in Widgets specifically its JS that does the rendering.

Here, I'll describe at a lower level in case one of us is misunderstanding something. I've contributed to NGLView on both the Python and JS side, so I think I have a working understanding, but it still feels very much like black magic:

Widgets are usually created in Python by subclassing `ipywidgets.Widget`, which provides Python and JS APIs to the `jupyterlab-widgets` and `widgetsnbextension` Jupyter extensions that are distributed with `ipywidgets`. The resulting `Widget` instance is synchronized by the extension with a widget model class (`WidgetModel`), written in JS. So if you change one of its special "traitlet" attributes in Python, the extension updates the same attribute in JS, and vice versa (assuming the traitlet has sync enabled). That's all a high level API over a message sending thingy that some widgets also use, but the general idea is that the widget and the model have the same data and you can define callbacks in either language to react to changes in that data. And then the model is represented in the notebook by a view (`WidgetView`), also written in JS, which does the rendering.

So rendering and interactivity is all written in JS, and that updates the shared data, and then Python is free to react to that (possibly changing the state). The result is that you can generally write responses to interactivity in either language, as long as you have both runtimes... running. And then `ipywidgets` includes some widgets that compose the others so it *looks* like you can basically write everything in Python... until you render everything to HTML and half your widgets break and you run off on a surprisingly successful wild goose chase to figure all this out.

Details: https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Low%20Level.html

I think my original paragraph accurately summarizes this, but it would be great to know how to make this clearer or where I've made a mistake!

Reply:
I'm not sure where my confusion was, but this all makes enough sense to me that it's well worth including as-is. Thanks for the explainer!